src/main/java/com/amazon/ion/impl/IonReaderContinuableCore.java

```diff
@@ -78,17 +79,31 @@ interface IonReaderContinuableCore extends IonCursor {
     int getFieldId();
 
     /**
-     * Gets the symbol IDs of the annotations attached to the current value.
+     * @return true if the value on which the reader is currently positioned has field
+     *  name text available for reading via {@link #getFieldText()}.
+     */
+    boolean hasFieldText();
+
+    /**
+     * Reads the text for the current field name. It is the caller's responsibility to
+     * ensure {@link #hasFieldText()} returns true before calling this method.
+     * @return the field name text.
+     */
+    String getFieldText();
+
+    /**
+     * Consumes SymbolTokens representing the annotations attached to the current value.
+     * Each SymbolToken provided will contain *either* a symbol ID, *or* its symbol
+     * text, depending on how it was encoded.
      * <p>
      * <b>This is an "expert method": correct use requires deep understanding
      * of the Ion binary format. You almost certainly don't want to use it.</b>
-     *
-     * @return the symbol IDs of the annotations on the current value.
-     * If the current value has no annotations, this method returns an empty array.
-     *
+     * <p>
+     * It is the caller's responsibility to ensure {@link #hasAnnotations()} returns
+     * true before calling this method.
```

Comment:
Is it easier to implement this way, or can we just say that the consumer will receive no values if there are no annotations?

Reply:
This is an internal API, so I'm happy to have a stricter contract that cuts down on extra branching.